src/components/Modal/hooks/useCalculateContentHeight.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useLayoutEffect, useState } from 'react';
+import { isNotNull } from 'ramda-adjunct';
+
+import { ContentHeightProps } from './useCalculateContentHeight.types';
+
+export const useCalculateContentHeight = (): ContentHeightProps => {
+  const [maxContentHeight, setMaxContentHeight] = useState(null);
+  const [modalHeight, setModalHeight] = useState(0);
+  const [headerHeight, setHeaderHeight] = useState(0);
+  const [footerHeight, setFooterHeight] = useState(0);
+
+  const onHeaderRefChange = useCallback((node) => {
+    if (isNotNull(node)) {
+      const { height } = node.getBoundingClientRect();
+      setHeaderHeight(height);
+    }
+  }, []);
+
+  const onModalRefChange = useCallback((node) => {
+    if (isNotNull(node)) {
+      const { height } = node.getBoundingClientRect();
+      setModalHeight(height);
+    }
+  }, []);
+
+  const onFooterRefChange = useCallback((node) => {
+    if (isNotNull(node)) {
+      const { height } = node.getBoundingClientRect();
+      setFooterHeight(height);
+    }
+  }, []);
```

Comment:
I don't think you need `useCallback` here since you are passing only internal variable you are using is `useState` setter which doesn't change during life cycle

Reply:
Removing this hook as agreed